Common/Source/MapDraw/DrawTerrain.cpp

```diff
@@ -533,9 +533,9 @@ class TerrainRenderer {
                             dd1 /= scale;
                             dd2 /= scale;
 
-                            int mag = (dd0 * dd0 + dd1 * dd1 + dd2 * dd2);
+                            int mag = isqrt4(dd0 * dd0 + dd1 * dd1 + dd2 * dd2);
                             if (mag > 0) {

```

Comment:
Doesnt isqrt4 always return a positive number? In such case we shall never use the rest of the code
} else {
						*imageBuf = oColorBuf[h];
					}
so something is wrong here.